sdk/core/azure_core/README.md

```diff
@@ -1,20 +1,283 @@
+
+### Consuming Long-Running Operations Using `Operation<T>`
+
+Some operations take long time to complete and require polling for their status. Methods starting long-running operations return `*Operation<T>` types.
+
+The `wait_for_completion` method is an easy way to wait for operation completion and get the resulting value.
+
+```rust no_run
+// create a client
+let client = SecretClient::new("http://example.com", DefaultAzureCredential::default(), Default::default());
+
+// Start the operation
+let mut operation = client.start_delete_secret("SecretName").await?;
+
+// Wait for the operation to complete
+let response = operation.wait_for_completion().await?;
+let value = response.value;
+
+println!("{}", value.name);
+println!("{:?}", value.scheduled_purge_date);
+```
+
+More on long-running operations in [long-running operation samples](https://github.com/Azure/azure-sdk-for-rust/blob/main/sdk/core/azure_core/samples/LongRunningOperations.md).
+
+### Customizing Requests Using `RequestContext`
+
+Besides general configuration of _service clients_ through `ClientOptions`, it is possible to customize the requests sent by _service clients_
+using protocol methods or convenience APIs that expose `RequestContext` as a parameter.
+
+```rust no_run
+let mut context = RequestContext::new();
+context.add_classifier(404, false);
+
+let response = client.get_pet("pet1", context).await?;
+```
+
+More on request customization in [RequestContext samples](https://github.com/Azure/azure-sdk-for-rust/blob/main/sdk/core/azure_core/samples/RequestContext.md).
+
+### Mocking
+
+One of the most important cross-cutting features of our new client libraries using `azure_core` is that they are designed for mocking.
+Mocking is enabled by:
+
+- providing a protected parameterless constructor on client types.
+- making service methods virtual.
+- providing APIs for constructing model types returned from virtual service methods. To find these factory methods look for types with the _ModelFactory_ suffix, e.g. `SecretModelFactory`.
+
+For example, the ConfigurationClient.get method can be mocked (with [Mockall](https://github.com/asomers/mockall)) as follows:
+
+```rust no_run
+// Create a mock response
+let mock_response = MockResponse::new();
+
+// Create a mock value
+let mock_value = SecretModelFactory::key_vault_secret(
+    SecretModelFactory::secret_properties("http://example.com".parse().unwrap())
+);
+
+// Create a client mock
+let mut mock = MockSecretClient::new();
+
+// Setup client method
+mock.expect_get_secret()
+    .with(eq("Name"), eq(None), eq(Default::default()))
+    .returning(move |_, _, _| Ok(Response::from_value(mock_value.clone(), mock_response.clone())));
+
+// Use the client mock
+let client = mock;
+let secret = client.get_secret("Name").await?;
+```
+
+More on mocking in [Unit testing and mocking with the Azure SDK for Rust](https://learn.microsoft.com/rust/azure/sdk/unit-testing-mocking).
+
+## Distributed tracing with OpenTelemetry
```

Comment:
This whole section is incorrect and we should probably just elide for now. We do support `tracing` which has an `opentelemetry` subscriber, but we haven't tested yet and don't actually have *distributed* tracing wired up correctly yet. I already have tracking issues open.

Reply:
Removed the section on tracing for now.